engine/src/main/java/org/terasology/rendering/world/WorldRendererImpl.java

```diff
@@ -539,6 +539,8 @@ private boolean isLightVisible(Vector3f positionViewSpace, LightComponent compon
 
     @Override
     public boolean isHeadUnderWater() {
+        // TODO: Making this as a subscribable value especially for node "ChunksRefractiveReflectiveNode",
+        // TODO: glDisable and glEnable state changes on that node will be dynamically added/removed based on this value.
```

Comment:
Hmm, well what I had in mind was not adding/removing nodes whenever player dives into to the water. I thought what if for instance ChunksRefractiveReflectiveNode is informed about this event and state changes added or removed according to this event.